### ISSUE TITLE
Add support for displaying C# code samples

### ISSUE THREAD

**Issue by MartinZikmund** [unoplatform/Uno.Gallery#487]
## What would you like to be added:

Currently we are able to display XAML code for samples, but not C# code-behind, which is especially relevant for non-UI APIs (but also for some UI APIs).

## Why is this needed:

Proper samples

## For which Platform:

All

**Comment by TopperDEL** [comment]
I have an idea for this and so would like to tackle this.

**Comment by TopperDEL** [comment]
Here a quick glimpse:
![image](https://user-images.githubusercontent.com/1833242/195383938-974077a4-883a-480b-84ce-f1209c5bb70b.png)

**Comment by TopperDEL** [comment]
Sleeping a night about that topic lead me to the conclusion that both topics deserve an own Nuget. The CodeHighlightTextBox might not be that important to put it into something like the Community toolkit, but it might be interesting for someone else. The same goes for the Source Generator to get the C#-code as a variable. This would clutter the gallery app.
So I'm going to create separate repos for both together with Nugets and use those Nugets in the gallery app.

**Comment by jeromelaban** [comment]
At this time, to have the best behavior, this would require the use of multiple instances of the Monaco editor, something that is not yet in a production-ready state (https://github.com/unoplatform/uno.monaco-editor-uwp/pull/14).

**Comment by TopperDEL** [comment]
That is a nice thing! But also quite heavy for just "showing some code". It is still the plan to use Monaco here? Simply showing code with syntax highlight, selectable and copy-able is already possible with the control from my spike.

So, how may I help then? We would still need a simple way to extract the code to show into a ViewModel-Property, right? Shall I concentrate on that or do you also have something in mind for that, too?

**Comment by MartinZikmund** [comment]
To have some kind of syntax highlighting and selectability, I would say a `WebView` with a JS syntax highlighter is still the easiest solution right now 🤔

**Comment by TopperDEL** [comment]
I had a working prototype for the Syntax-highlighting (see above). That could be used and is purely C#-stuff. And we would still need a way to extract the code from the sample-code without doubling it (at least this would be nice).

If we want to go that route I would still be happy to contribute. :)

**Comment by MartinZikmund** [comment]
@TopperDEL your approach uses WebView under the hood correct?

**Comment by TopperDEL** [comment]
No, it is a pure C#-implementation. I found a TextBlock-project which renders text with syntax-highligting. See [here](https://github.com/ryanvalentin/UWP-RichTextControls).

**Comment by MartinZikmund** [comment]
@TopperDEL  @morning4coffe-dev the issue here will be mainly the lack of `RichTextBlock` support. I would say using an embedded `GitHub` code could be easier. On mobile targets and UWP we could just use `WebView` for that, on WASM we could for now have a custom control, that renders the GitHub `<iframe>`. Nice thing would also be that people could then navigate to the snippet on GitHub directly

**Comment by francoistanguay** [comment]
I feel supporting multiple Monaco instances should be the way to go and we'll most probably need it in a refreshed playground...

**Comment by Xiaoy312** [comment]
I remember there was issue with having multiple monaco editors on wasm, where we cant target an specific editor to set content; it just always target the first. In figma, we got around it by reusing the same editor instance, but at the cost of performance (changing "editor"/tab = reloading the content). The same tech can be used for playground, but it wouldnt work in gallery, due the way it is setup: we cant move the editor around on the visual tree, only resizing. We should probably address this problem first.

Syntax highlighting aside, I think there is also the automatic retrieval of actual source, sort of like what ShowMeTheXaml does for .xaml file, but for .cs.

**Comment by agneszitte** [comment]
We should match a bit more what WinUI gallery is doing I think

- Current Uno.Gallery version (XAML + C#)
	
![image](https://github.com/unoplatform/Uno.Gallery/assets/16295702/f9c22e21-a282-48d2-b7e7-ae4a258eeb74)
	
- WinUI Gallery version (XAML + C#)

![image](https://github.com/unoplatform/Uno.Gallery/assets/16295702/890e9942-2f2d-4bd9-8089-bc093ddc40f8)

**Comment by MartinZikmund** [comment]
I think the easiest could be to use a `WebView` and embed GitHub gists or other source code embedding. This way the same UI will show everywhere
